### PR TITLE
Rust!

### PR DESCRIPTION
**Pull request by quad**
I've ported exit to rust fairly literally.

# Open Questions

- How do you feel about the use of Hermit?
- `from_signal` returns `i32` because there aren't enumerated values for signals; should we add them?

### CODE REVIEW

**Comment by boblail** [comment]
> How do you feel about the use of Hermit?

I'm for it! Two questions:

- If this repo has been hermitized, I think using hermit would still be optional for contributors, is that correct?
   i.e. I'd be reluctant to add a dependency to something this simple
- We would use hermit in two [forthcoming] places, I'm assuming?
    1. a GitHub Action that runs the tests
    2. a section of the README about Contributing

**Comment by boblail** [comment]
> `from_signal` returns `i32` because there aren't enumerated values for signals; should we add them?

How often does the list of signals change? Is it fairly portable?

I read [signal(7)](https://man7.org/linux/man-pages/man7/signal.7.html) and it seems like the answers are "Not often" and the list is basically the same of Intel and Arm processors.

Can realtime signals cause a program to terminate? If so, does it still terminate with 128 + `N`?

* * *

In the Go code, I made the type an alias to `int` instead of a "real" enum so that the exit codes could be passed directly to `os.Exit`. I don't _think_ the Go code would be much improved by enumerating the signals, but if the Rust implementation would be — I'd take your word for it! 😃

**Review by boblail — APPROVED** [review]

**Comment by quad** [comment]
> > How do you feel about the use of Hermit?
> 
> I'm for it! Two questions:
> 
> * If this repo has been hermitized, I think using hermit would still be optional for contributors, is that correct?
>   (i.e. I'd be reluctant to add a dependency to something this simple)

Hermit will be optional!

> * We would use hermit in two [forthcoming] places?
>   
>   1. a GitHub Action that runs the tests
>   2. a section of the README about Contributing

We could!

I saw neither a test action nor a part in the README that said how to run things so I didn't include anything. 😅

**Comment by quad** [comment]
> > `from_signal` returns `i32` because there aren't enumerated values for signals; should we add them?
> 
> How often does the list of signals change? Is it fairly portable?

Fantastic question! I have no idea!

Naively I'd assume that the older UNIX signals are all set in stone, where SYSV and BSD would have diverged, and Linux would have its own too…

> I read [signal(7)](https://man7.org/linux/man-pages/man7/signal.7.html) and it seems like the answers are "Not often" and the list is basically the same of Intel and Arm processors.

Whoa, I wouldn't have expected them to change based on processor architecture!

> 
> Can realtime signals cause a program to terminate? If so, does it still terminate with 128 + `N`?

Realtime signals are beyond my ken.

> In the Go code, I made the type an alias to `int` instead of a "real" enum so that the exit codes could be passed directly to `os.Exit`. I don't _think_ the Go code would be much improved by enumerating the signals, but if the Rust implementation would be — I'd take your word for it! 😃

Rust is a _little_ more strict about its type aliases. 🔝 I might add an "examples" directory?

**Comment by boblail** [comment]
> I might add an "examples" directory?

👍  Sure! I'm OK with that!

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "semantic-exit"
```
Comment (quad) [inline Cargo.toml:2]: @boblail OK with this crate name?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,8 +2,8 @@ on: push
 
 name: CI
 jobs:
-  test:
-    name: Test
+  go:
+    name: Test (Go)
```
Comment (quad) [inline .github/workflows/ci.yml:6]: @boblail I changed the CI workflow name for the Go test

File: Cargo.toml (hunk 1 above)
Comment (boblail) [reply Cargo.toml:2]: Yep!

**Pull request merged**